lib/firmenwissen/http_request.rb

```diff
@@ -8,7 +8,7 @@ def initialize(uri, options = {})
     def execute
       http = Net::HTTP.start(uri.host, uri.port, use_ssl: uri.use_ssl?)
       http.read_timeout = config.timeout
-      http.request(request)
+      http.request(request, &method(:extract_session))
```

Comment:
I'd prefer actual block syntax here, but that might only be due to my limited knowledge about Net::HTTP (I had to check the docs what this block is called with, after that it was fine for me)

Reply:
I found this version to be less distracting from the core mission of `execute`. During implementation I thought about using a more general name for the callback, e.g. `preprocess_response`. Would this be more in your favor?

Reply:
I guess so, yes. But feel free to keep it as is, it's not bothering me really :)